src/main/java/org/openrewrite/java/testing/assertj/testng/TestNgAssertEqualsToAssertThat.java

```diff
@@ -0,0 +1,150 @@
+        private JavaParser.Builder<?, ?> assertionsParser;
+
+        private JavaParser.Builder<?, ?> assertionsParser(ExecutionContext ctx) {
+            if (assertionsParser == null) {
+                assertionsParser = JavaParser.fromJavaVersion()
+                        .classpathFromResources(ctx, "assertj-core-3.24");
+            }
+            return assertionsParser;
+        }
+
+        private static final MethodMatcher TESTNG_ASSERT_EQUALS = new MethodMatcher("org.testng.Assert" + " assertEquals(..)");
+
+        @Override
+        public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
+            if (!TESTNG_ASSERT_EQUALS.matches(method)) {
+                return method;
+            }
+
+            List<Expression> args = method.getArguments();
+            Expression expected = args.get(1);
+            Expression actual = args.get(0);
+
+            //always add the import (even if not referenced)
+            maybeAddImport("org.assertj.core.api.Assertions", "assertThat", false);
+
+            // Remove import for "org.testng.Assert" if no longer used.
+            maybeRemoveImport("org.testng.Assert");
+
+            if (args.size() == 2) {
+                return JavaTemplate.builder("assertThat(#{any()}).isEqualTo(#{any()});")
+                        .staticImports("org.assertj.core.api.Assertions.assertThat")
+                        .javaParser(assertionsParser(ctx))
+                        .build()
+                        .apply(getCursor(), method.getCoordinates().replace(), actual, expected);
+            } else if (args.size() == 3 && !isFloatingPointType(args.get(2))) {
+                Expression message = args.get(2);
+                JavaTemplate.Builder template = TypeUtils.isString(message.getType()) ?
+                        JavaTemplate.builder("assertThat(#{any()}).as(#{any(String)}).isEqualTo(#{any()});") :
+                        JavaTemplate.builder("assertThat(#{any()}).as(#{any(java.util.function.Supplier)}).isEqualTo(#{any()});");
+                return template
+                        .staticImports("org.assertj.core.api.Assertions.assertThat")
+                        .imports("java.util.function.Supplier")
+                        .javaParser(assertionsParser(ctx))
+                        .build()
+                        .apply(
+                                getCursor(),
+                                method.getCoordinates().replace(),
+                                actual,
+                                message,
+                                expected
+                        );
+            } else if (args.size() == 3) {
+                //always add the import (even if not referenced)
+                maybeAddImport("org.assertj.core.api.Assertions", "within", false);
+                return JavaTemplate.builder("assertThat(#{any()}).isCloseTo(#{any()}, within(#{any()}));")
```

Comment:
for
`assertEquals(digest.valueAt(0), value, 0.001f);`
it generates
`assertThat(digest.valueAt(0)).isCloseTo(value, within(0.001f));`
which does not compile, because previously float was implicitly converted to double
It has to be 
`assertThat(digest.valueAt(0)).isCloseTo(value, within(0.001));`
https://github.com/airlift/airlift/pull/1194/files#diff-f3a36adcfe5a7f0cf6f257f92ac316bb103c8cce86a3850b1e343132dc5ec633L148